src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java

```diff
@@ -260,12 +260,22 @@ private boolean isBlankAddress(final PrisonContactDetail pcd) {
     public List<OffenderCell> getCellsWithCapacityInAgency(@NotNull final String agencyId, String attribute) {
         final var livingUnits = livingUnitRepository.findAllByAgencyLocationId(agencyId);
         return livingUnits.stream()
-                .map(this::transform)
+                .map(livingUnit -> transform(livingUnit, true))
                 .filter(Objects::nonNull)
                 .filter(cell -> attribute == null || cell.getAttributes().stream().map(OffenderCellAttribute::getCode).collect(toList()).contains(attribute))
                 .collect(toList());
     }
 
+    @Override
+    public OffenderCell getCellAttributes(@NotNull final Long locationId) {
+        final var livingUnit = livingUnitRepository.findOneByLivingUnitId(locationId);
+        final var offenderCell = livingUnit.map(unit -> transform(unit, false)).orElse(null);
```

Comment:
You might be able to do a `orElseThrow (new  EntityNotFoundException.withMessage(String.format("No cell details found for location id %s", locationId)))`

Reply:
There's another null possible as a result of the transform, so didn't work neatly with the above.